crates/editor/src/element.rs

```diff
@@ -3889,11 +3886,14 @@ impl ScrollbarLayout {
 
         let mut quads = Vec::new();
         while let Some(mut pixel_range) = background_pixel_ranges.next() {
+            pixel_range.end = pixel_range
+                .end
+                .max(pixel_range.start + Self::MIN_MARKER_HEIGHT);
```

Comment:
Why is the code in here? I think the original location, where `end_y` is constructed` is better. That way we construct the range in one place and iterate over it in another.

Reply:
I think it's better to place it here because the min height must be applied to the resulting aggregated quad, not to each marker. If we put it higher (as it is now), we always extend quad even when it's not necessary (when the resulting quad is already higher than the min height).

Here's an example that illustrates the idea:

For simplicity, let's assume we have a file with rows number equal to the scrollbar height, i.e. "pixels per row" = "rows per pixel" = 1. And let's assume rows 1 and 2 contain highligted search results.

In this case, without adjusting to the min marker height, we have initial quad ranges [1, 1] and [2, 2]. Thus, the aggregated quad could be [1, 2], which is a) absolutely accurate, b) satisfies the "min height = 2px" rule.

The current implementation imposes min height to every quad before aggregation, transforming them to [1, 2] and [2, 3]. After merging, we end up with the single quad [1, 3], which is a) not accurate, b) unnecessarily high.

With this PR we have quad 1 starting with pixels range [1, 2] (we make sure we'll never end up with a too narrow quad), but the second quad [2, 2] won't expand it. So we come back to the accurate result [1, 2].

Reply:
Thanks for that explanation. I think that makes sense.